internal/acctest/template/field_extractor.go

```diff
@@ -0,0 +1,277 @@
+package template
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+
+	datasourceschema "github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	resourceschema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	sdkschema "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// SortTemplateFields returns a sorted copy of the fields slice to ensure deterministic rendering
+func SortTemplateFields(fields []TemplateField) []TemplateField {
```

Comment:
nit: I would think to incapsulate these things together with other types here inside a package. Looks to me like internal implementation details.